### PR TITLE
Feature lwdev 8828 cancell all limit orders endpoint

### PR DESCRIPTION
**Pull request by nikagamkrelidze**

### CODE REVIEW

File: LykkeApi2/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -266,7 +270,82 @@ public async Task<IActionResult> PlaceLimitOrder([FromBody] LimitOrderRequest re
...
+            }
+
+            var clientId = _requestContext.ClientId;
+            
+            var activeOrders = await _limitOrdersRepository.GetActiveByClientIdAsync(clientId);
+
+            if (!activeOrders.Any())
+                return NotFound();
+
+            await _limitOrdersRepository.CancelMultipleAsync(
```
Comment (polonzer) [inline LykkeApi2/Controllers/OrdersController.cs]: Weird code. There can be inconsistency problems. But if it is an approved architecture decision, ok.

Comment (polonzer) [reply LykkeApi2/Controllers/OrdersController.cs]: Yes, there is no need to call it here. TxHandler will do the same things after ME executes the command.

File: LykkeApi2/Controllers/OrdersController.cs [hunk 2]
```diff
@@ -266,7 +270,82 @@ public async Task<IActionResult> PlaceLimitOrder([FromBody] LimitOrderRequest re
...
+            }
+
+            var clientId = _requestContext.ClientId;
+
+            var activeOrders = await _limitOrdersRepository.GetActiveByClientIdAsync(clientId);
+
+            if (activeOrders.All(x => x.Id != orderId))
+                return NotFound();
+            
+            await _limitOrdersRepository.CancelByIdAsync(clientId, orderId);
```
Comment (polonzer) [inline LykkeApi2/Controllers/OrdersController.cs]: There is no need to call it here. TxHandler will do the same things after ME executes the command.

File: LykkeApi2/Controllers/OrdersController.cs [hunk 3]
```diff
@@ -266,7 +270,82 @@ public async Task<IActionResult> PlaceLimitOrder([FromBody] LimitOrderRequest re
...
 
             return Created(Url.Action("Get", "Operations", new { id }), id);                  
-        }        
+        }
+        
+        [HttpDelete("limit/{orderId}")]
+        [SwaggerOperation("CancelLimitOrderNew")]
+        [ProducesResponseType(typeof(void), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(void), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> CancelLimitOrderNew(string orderId)
```
[the remark this answers is not in the document]
Comment (nikagamkrelidze) [reply LykkeApi2/Controllers/OrdersController.cs]: this is a new endpoint, the old one will be removed as soon as this is released.

File: LykkeApi2/Controllers/OrdersController.cs [hunk 4]
```diff
@@ -266,7 +270,82 @@ public async Task<IActionResult> PlaceLimitOrder([FromBody] LimitOrderRequest re
             }
 
             return Created(Url.Action("Get", "Operations", new { id }), id);                  
-        }        
+        }
+        
+        [HttpDelete("limit/{orderId}")]
```
[the remark this answers is not in the document]
Comment (nikagamkrelidze) [reply LykkeApi2/Controllers/OrdersController.cs]: contracts were discussed with the core team, this is what we came up with. i think it's important to specify that only limit orders can be canceled.

File: LykkeApi2/Controllers/OrdersController.cs [hunk 5]
```diff
@@ -266,7 +270,82 @@ public async Task<IActionResult> PlaceLimitOrder([FromBody] LimitOrderRequest re
...
+            if (activeOrders.All(x => x.Id != orderId))
+                return NotFound();
+            
+            await _limitOrdersRepository.CancelByIdAsync(clientId, orderId);
+            await _matchingEngineClient.CancelLimitOrderAsync(orderId);
+            
+            return Ok();
+        }
+
+        [HttpDelete("limit")]
```
[the remark this answers is not in the document]
Comment (nikagamkrelidze) [reply LykkeApi2/Controllers/OrdersController.cs]: same here.

**Pull request merged**